pkg/syncer/k8scloudoperator/placement.go

```diff
@@ -624,20 +626,28 @@ func eliminateNodesWithPvcOfSiblingReplica(ctx context.Context, client kubernete
 		}
 
 		pvName := pvc.Spec.VolumeName
-		pv, err := client.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
-		if err != nil || pv == nil {
-			if pvName == "" {
-				continue
-			}
-			return candidateHosts, err
+		if pvName == "" {
+			continue
 		}
 
-		// Volume Binding has to be WaitForFirstConsumer because it can only be
-		// Immediate if it has the annotation failure-domain.beta.vmware.com/node
-		// on the PVC. We have already verified that this annotation is not
-		// present on PVC as part of the validatin above. Hence, nodeAffinity
-		// value will always be defined on the PV.
-		hostName := pv.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
+		hostName := ""
+		if val, ok := pvc.Annotations[pvcSelectedNode]; ok {
+			log.Infof("PVC has selected node annotation %+v", currPVC.Name)
+			hostName = val
+		} else {
+			log.Infof("Fetching PV %+v", currPVC.Name)
+			pv, err := client.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+			if err != nil || pv == nil {
+				return candidateHosts, err
+			}
+
+			// Volume Binding has to be WaitForFirstConsumer because it can only be
+			// Immediate if it has the annotation failure-domain.beta.vmware.com/node
+			// on the PVC. We have already verified that this annotation is not
+			// present on PVC as part of the validatin above. Hence, nodeAffinity
+			// value will always be defined on the PV.
+			hostName = pv.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
```

Comment:
Can you confirm if all of the above array indices have items at all times ?
Can `Values` or `MatchExpressions` or `NodeSelectorTerms` be empty ?

Reply:
Can you unit test this function `eliminateNodesWithPvcOfSiblingReplica` with a sample/mocked PV Spec ?

Reply:
> Can you confirm if all of the above array indices have items at all times ?
> Can Values or MatchExpressions or NodeSelectorTerms be empty ?

Like I have mentioned in the comment right above this one,  nodeaffinity value must always be populated as the volume binding is WFFC. But I have also added more validations to avoid any failures.

> Can you unit test this function eliminateNodesWithPvcOfSiblingReplica with a sample/mocked PV Spec ?
Currently we don't have any unit test cases for placement engine. We have an action item on the same already. Do you want me to add a unit test anyway?